robottelo/hosts.py

```diff
@@ -1842,14 +1843,15 @@ def get_caller():
                 if frame.function.startswith('test_'):
                     return frame.function
 
-        return Session(
+        self._ui_session = Session(
             session_name=testname or get_caller(),
             user=user or settings.server.admin_username,
             password=password or settings.server.admin_password,
             url=url,
             hostname=self.hostname,
             login=login,
         )
+        return self._ui_session
```

Comment:
This patch is not necessary then, correct?

Reply:
Nice Catch!